Remove `--no-default-features` from uv formula

zanieb opened the pull request:
This disables the `performance` feature, which provides critical performance improvements for production use-cases

carlocab reviewed (commented):
> Using this flag disables the `performance` feature, which provides critical performance improvements for production use-cases.

Do you have benchmarks for these improvements?

zanieb commented:
I can generate some benchmarks, but we rigorously tested all of the performance optimizations included there when we added them in the first place. For quite some time (uv 0.1.0 -> 0.4.17), these were _already_ enabled in Homebrew, until we moved them to a feature to allow us to toggle them off for development builds.

zanieb commented:
Separately from the argument that this will improve performance (which I think may only broadly apply to pathological cases we've optimized), we do not intend downstream consumers to build without the default features and it is quite feasible disabling the default features like this will lead to additional unexpected regressions in the future.

konstin commented on Formula/u/uv.rb:
```diff
@@ -26,7 +26,7 @@ class Uv < Formula
     ENV["UV_COMMIT_HASH"] = ENV["UV_COMMIT_SHORT_HASH"] = tap.user
     ENV["UV_COMMIT_DATE"] = time.strftime("%F")
-    system "cargo", "install", "--no-default-features", *std_cargo_args(path: "crates/uv")
+    system "cargo", "install", *std_cargo_args(path: "crates/uv")
```
Added some clarifications on the features in https://github.com/astral-sh/uv/pull/13265

zanieb commented:
Our standard benchmarks actually don't show a major difference here, which is reassuring. However, I do believe there are pathological cases where the performance features are relevant. Additionally, we expect these features to be used in production and it could easily be confusing for us trying to help users.

As an upstream maintainer, I strongly urge you to remove the `--no-default-features` flag as it causes unnecessary differences in the distributions of uv and future maintenance burden.

zanieb commented:
👋 please don't stale bot this pull request

zanieb commented:
cc @chenrui333 (you reviewed the original formula)

branchvincent commented:
thank you @zanieb, sorry for the delay. this will ship with the next `uv` release

zanieb commented:
Thank you!
